apps/judicial-system/backend/src/app/modules/police/police.service.ts

```diff
@@ -477,38 +482,62 @@ export class PoliceService {
-              sveitafelag?: string | null
-            }) => {
-              const policeCaseNumber = info.upprunalegtMalsnumer
-
-              const place = formatCrimeScenePlace(
-                info.gotuHeiti,
-                info.gotuNumer,
-                info.sveitafelag,
-              )
-              const date = info.brotFra ? new Date(info.brotFra) : undefined
-              const licencePlate = info.licencePlate
-
-              const foundCase = cases.find(
-                (item) => item.policeCaseNumber === policeCaseNumber,
-              )
-
-              if (!foundCase) {
-                cases.push({ policeCaseNumber, place, date, licencePlate })
-              } else if (date && (!foundCase.date || date > foundCase.date)) {
-                foundCase.place = place
-                foundCase.date = date
-                foundCase.licencePlate = licencePlate
-              }
-            },
+          await Promise.all(
+            (response.malseinings ?? []).map(
+              async (info: {
+                upprunalegtMalsnumer: string
+                vettvangur?: string
+                brotFra?: string
+                licencePlate?: string
+                gotuHeiti?: string | null
+                gotuNumer?: string | null
+                sveitafelag?: string | null
+                artalNrGreinLidur?: string | null
+              }) => {
+                const policeCaseNumber = info.upprunalegtMalsnumer
+                const article = info.artalNrGreinLidur
+                const subtype = await this.getSubtypeByArticle(article)
+                const key = Object.keys(IndictmentCaseSubtypes).find(
+                  (k) =>
+                    IndictmentCaseSubtypes[
+                      k as keyof typeof IndictmentCaseSubtypes
+                    ] === subtype?.offenseType,
+                )
+
+                const place = formatCrimeScenePlace(
+                  info.gotuHeiti,
+                  info.gotuNumer,
+                  info.sveitafelag,
+                )
+                const date = info.brotFra ? new Date(info.brotFra) : undefined
+                const licencePlate = info.licencePlate
+
+                const foundCase = cases.find(
+                  (item) => item.policeCaseNumber === policeCaseNumber,
+                )
+
+                if (!foundCase) {
+                  cases.push({
+                    policeCaseNumber,
+                    place,
+                    date,
+                    licencePlate,
+                    subtypes: key ? [key] : [],
+                  })
+                } else {
+                  if (date && (!foundCase.date || date > foundCase.date)) {
+                    foundCase.place = place
+                    foundCase.date = date
+                    foundCase.licencePlate = licencePlate
+                    foundCase.subtypes = []
+                  }
+
+                  if (key && !foundCase.subtypes?.includes(key)) {
```

Comment:
Nit: We can also create a set which we insert into, then we don´t need to check if the found case includes the key.